rest/src/main/kotlin/builder/interaction/OptionsBuilder.kt

```diff
@@ -58,36 +65,66 @@ public sealed class BaseChoiceBuilder<T>(
     private var _choices: Optional<MutableList<Choice<*>>> = Optional.Missing()
     public var choices: MutableList<Choice<*>>? by ::_choices.delegate()
 
-    public abstract fun choice(name: String, value: T)
+    public abstract fun choice(name: String, value: T, nameLocalizations: Optional<Map<Locale, String>>)
+
+    /**
+     * Registers a new choice with [name] representing value and applies [localizationsBuilder] to it
+     *
+     * @see ChoiceLocalizationsBuilder
+     */
+    public inline fun choice(name: String, value: T, localizationsBuilder: ChoiceLocalizationsBuilder.() -> Unit = {}) {
+        val localizations = ChoiceLocalizationsBuilder(name)
+            .apply(localizationsBuilder)
+            .nameLocalizations
+            ?: emptyMap()
+        val optional = if (localizations.isEmpty()) {
+            Optional.Missing()
+        } else {
+            Optional(localizations)
+        }
+
+        return choice(name, value, optional)
+    }
 
     override fun toRequest(): ApplicationCommandOption = ApplicationCommandOption(
         type,
         name,
+        _nameLocalizations,
         description,
+        _descriptionLocalizations,
         choices = _choices,
         required = _required,
         default = _default,
         autocomplete = _autocomplete,
     )
 }
 
+/**
+ * Builder to register name localizations for a choice.
+ *
+ * @see LocalizedNameBuilder
+ */
+public class ChoiceLocalizationsBuilder(override val name: String?) : LocalizedNameBuilder {
+    override var nameLocalizations: MutableMap<Locale, String>? = mutableMapOf()
+}
```

Comment:
```suggestion
public class ChoiceLocalizationsBuilder(override var name: String) : LocalizedNameCreateBuilder {
    @PublishedApi
    internal var _nameLocalizations: Optional<MutableMap<Locale, String>> = Optional.Missing()
    override var nameLocalizations: MutableMap<Locale, String>? by ::_nameLocalizations.delegate()
}
```
`LocalizedNameCreateBuilder` instead of `LocalizedNameBuilder` -> mutable, non-nullable `name`
delegate to internal property